### DISCUSSION TITLE
Install and use several versions without "relinking"

### DISCUSSION THREAD

**Discussion by aywan** [shivammathur/homebrew-php#1768] [Q&A / Help]
Hello, I'm absolutely newbie in macos)

On Linux Ubuntu I used to ondrej ppa with several versions of Php. After install you can use any version without linking process. For example when I want use 8.2 - i type in console `php8.2`, and `php7.4` for 7.4)
I install several versions, but can't found it binaries for make symlink into bin dir. I think that /opt/homebrew/bin/php is just symlink, but it is binary.
Maybe it's good idea to provide access to all installed versions simultaneously without relinking?

**Comment by shivammathur** [comment]
@aywan 

You can use different PHP versions in macOS as well without linking, but you will have to update your paths each month when a new php semver version is released.

both apt packages on Ubuntu, and brew packages on macOS have linking.

- On Ubuntu
It installs the binaries by default in `/usr/bin`. For example `/usr/bin/php8.1`, `/usr/bin/php8.2`, etc,
There is also a symlink `/usr/bin/php` that links to `/etc/alternatives/php`. `/etc/alternatives/php` links to the latest php binary installed by default, and you can switch using the `update-alternatives` command.

- On macOS using brew
It installs packages in a folder called `Cellar`. In your case, it will be `/opt/homebrew/Cellar`. Inside that, you can find directories for all the brew packages that are installed. Then, for example, the path for PHP 8.1 binary can be `/opt/homebrew/Cellar/php@8.1/8.1.20/bin/php`.
Please notice that the exact php semver version is in this path, brew provides functionality to link it to a directory called `prefix`. In your case it is `/opt/homebrew` and the bin directory in the prefix would be `/opt/homebew/bin` where it will link the binary from the cellar directory.

So, packages on both OS have linking, ubuntu replaces the binaries when there is a new semver version, so it does not have it in the path of the binaries. brew keeps the old semver version and installs the new semver version in the cellar.
